Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1524,9 +1524,18 @@ void RenderTheme::adjustSliderThumbStyle(RenderStyle& style, const Element* elem
     adjustSliderThumbSize(style, element);
 }
 
-void RenderTheme::adjustSwitchStyle(RenderStyle& style, const Element* element) const
+void RenderTheme::adjustSwitchStyle(RenderStyle& style, const Element*) const
 {
-    adjustButtonOrCheckboxOrColorWellOrInnerSpinButtonOrRadioOrSwitchStyle(style, element);
+    auto zoomedSize = style.isHorizontalWritingMode() ? LengthSize { style.width(), style.height() } : LengthSize { style.height(), style.width() };
```

Comment:
This looks wrong. What if `style.width()` or `style.height()` are not fixed values.

Need to check `.isFixed()`. Also we have `logicalWidth()` and `logicalHeight()` to avoid checking `style.isHorizontalWritingMode()` here.

Reply:
`switchSize()` in `ThemeMac` handles this.

Reply:
I double checked the code in `adjustButtonOrCheckboxOrColorWellOrInnerSpinButtonOrRadioOrSwitchStyle()` btw and I feel more confident now that this is identical to what it did before (except now accounting for "logical" and with much less no-op code).